avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1522,6 +1522,12 @@ private Schema toConnectSchema(org.apache.avro.Schema schema,
     AvroSchemaAndVersion schemaAndVersion = new AvroSchemaAndVersion(schema, version);
     Schema cachedSchema = toConnectSchemaCache.get(schemaAndVersion);
     if (cachedSchema != null) {
+      if (schema.getType() == org.apache.avro.Schema.Type.RECORD) {
```

Comment:
Not sure we need it, but maybe add `if (schema != null && schema.getType() == ...`

Reply:
Should not be necessary, if the schema was null `toConnectSchema` below would have crashed before this and nothing else populates `toConnectSchemaCache`.